Configure

```diff
@@ -1451,6 +1451,13 @@ if (!$disabled{asm}) {
     }
 }
 
+if (!$disabled{ec_nistp_64_gcc_128} && !$predefined{__SIZEOF_INT128__}) {
```

Comment:
If we can detect its presence at Configure time, could we enable it if present (unless explicitly disabled)?

Reply:
One should keep in mind that availability of `__SIZEOF_INT128__` is not the only requirement for the code in question. Target platform has to be little-endian and tolerate misaligned memory references.

Reply:
It should be noted that it prevents user from passing `-D__SIZEOF_INT128__=16`. Though I recognize that one can argue that doing this should not be an option.

Reply:
Re little endian and misaligned tolerance, I see no evidence of such requirements in `ecp_nistp224.c` or `ecp_nistp256.c`.  I *do* see such requirements in `curve25519.c`, though, but that one isn't affected by the `ec_nistp_64_gcc_128` option anyway.  What am I missing?

Reply:
>  Re little endian and misaligned tolerance, I see no evidence of such requirements

See bin28_to_felem in ecp_nistp224.c for example.

> I do see such requirements in curve25519.c

If there was, how would it work on big-endian system?

Reply:
> See bin28_to_felem in ecp_nistp224.c for example.

Aha, ok, implicit requirements...  I though you were talking about things we guard against.  But ok, I see your argument against enabling this by default without checking for the things we assume.

Reply:
> Target platform has to be little-endian and tolerate misaligned memory references.

That's news to me and isn't documented anywhere AFAICT. It should probably at least have a note about this in the "INSTALL" doc (as part of the existing documentation on that option).

Reply:
Well, there is that red flag, "This is only supported on some platforms." ...

Reply:
(I read that as "you're welcome to try it, but we leave *no* guarantees whatsoever that it'll work for you")

Reply:
INSTALL updated